### ISSUE TITLE
Question: Is `avoid_single_child` rule producing a false positive for collection for-loops?

### ISSUE THREAD

**Issue by kosukesaigusa** [altive/altive_lints#51]
I've been using the altive_lints package with `all_lint_rules.yaml` and recently started using the altive recommended rules. Thank you for this great package!

I've noticed that the `avoid_single_child` rule seems to flag a warning for `ListView` `children` when using a collection for-loop. I'd like to ask your opinion on whether this is intended behavior or a potential false positive.

Here's an example that triggers the lint:

```dart
ListView(
  children: [
    for (final todo in todos)
      ListTile(title: Text(todo.title)),
  ],
),
```

My understanding is that a collection for-loop can potentially generate multiple children, so this might not be a case where we're actually using a single child.

What are your thoughts on this? Do you consider this a false positive, or is there a reason for this behavior that I'm missing?

Environment:
- altive_lints version: 1.11.1
- Dart version: 3.4.3
- Flutter version: 3.22.2

Thank you for your time and consideration!

**Comment by naipaka** [comment]
@kosukesaigusa 
Thank you for your question and for using the package!

You're absolutely right—this behavior is indeed a false positive. 

We're aware of this issue and plan to address it soon.
You can track the progress on this at https://github.com/altive/altive_lints/issues/42. 
We're also planning to address issues related to if conditions along with the for loop fix.

Thank you for your patience as we work on this.

**Comment by kosukesaigusa** [comment]
Thank you for your kind and prompt response!

I'm sorry, I should have confirmed the issue already created!
